SPECS/onefuzz/onefuzz.spec

```diff
@@ -0,0 +1,54 @@
+Summary:        A self-hosted Fuzzing-As-A-Service platform
+Name:           onefuzz
+Version:        8.5.0
+Release:        1%{?dist}
+License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Development/Tools
+URL:            https://github.com/microsoft/OneFuzz
+Source0:        https://github.com/microsoft/onefuzz/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
+Patch0:         git-version.patch
+
+BuildRequires:  cargo >= 1.68
+BuildRequires:  git
+BuildRequires:  kernel-headers
+BuildRequires:  libunwind-devel
+BuildRequires:  openssl-devel
+BuildRequires:  perl
+BuildRequires:  rust >= 1.68
+BuildRequires:  util-linux
+BuildRequires:  xz-devel
+
+%description
+Project OneFuzz enables continuous developer-driven fuzzing to proactively
+harden software prior to release. With a single command, which can be baked
+into CICD, developers can launch fuzz jobs from a few virtual machines to
+thousands of cores.
+
+%prep
+%autosetup -p1
+
+%build
+cd src/agent
+ONEFUZZ_SET_VERSION=%{version} RUSTFLAGS="-llzma" cargo build --release
```

Comment:
@cswindle
On further inspection, there is an edge case which we are hitting here as per the design of AzureLinux build system.
- The toolkit does not allow internet access during a normal build; but how did the local build succeed you might ask; Well, if we build with RUN_CHECK=y (which is only used for tests) Internet access is enabled.

Maybe we could sync-up offline for further discussions. I did manage to build it locally (with RUN_CHECK=y [internet enabled]), but for offline building we might want to vendor sources by doing `cargo vendor` and caching the entire sources that it has to download from the internet.

I did some slight modifications to the BuildRequires, 
```
- BuildRequires: kernel-headers
+ BuildRequires: perl(FindBin)
+ BuildRequires: perl(lib)
```